### ISSUE TITLE
automatically log exceptions that bubble up to the top level

### ISSUE THREAD

**Issue by lloyd** [mozilla/chromeless#36]
as a convenience to developers, it would be useful if exceptions that bubble to the top level were automatically logged to the console, as well as html syntax errors.

there must be a way to hook into gecko at a low level and get our code running when these events occur...

**Comment by taboca** [comment]
Mike's commit fixes this? I noticed the ./tests/console/index.html is broken when launching the app separated mode ( the newer approach which is similar to --install-app )

**Comment by lloyd** [comment]
Aha!  Yeah, interesting find.  Here's the explanation:

window.console.log (which we inject) is set up by the harness.  it will always output to the true application console, but in development mode it will also output to a temporary file that we'll pipe to the true interactive user's console.

on the other hand, plain-text-console will only output to the true program console.  

I've updated the test to demonstrate this behavior:
https://github.com/mozilla/chromeless/commit/d6eb8f9d96bbd8251812feffe17e93906f2aa961

I think we might want to consider removing plain-text-console, because I feel like it's redundant and perhaps confusing.  More generally I think we spend some time restructuring all of the modules into a better package hierarchy to make things easier to find.  I feel like with a little thought we could make what's becoming our "standard library" a lot more developer friendly.

Closing this issue, because Mikes commit does fix what this issue is about, uncaught javascript exceptions now make it to the console.log, but I think we can open up several little issues now to refine the feature and maybe kill that redundancy..
